### PR TITLE
test(pkg/version): migrate test-infra to testify for pkg/version pkg

### LINKED ISSUE #2919: migrate test-infra to testify for `pkg/version` pkg

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve? 
migrate test-infra to testify for pkg/version pkg, close  #2919

### What is changed and how it works?

1. migrate test-infra from gocheck to testify
2. migrate leak test from leaktest to goleak

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
migrate test-infra to testify for pkg/version pkg
```

### CODE REVIEW

File: pkg/version/check_test.go [hunk 1]
```diff
@@ -18,26 +18,19 @@ import (
...
 	"testing"
 	"time"
 
 	"github.com/coreos/go-semver/semver"
-	"github.com/pingcap/check"
 	"github.com/pingcap/kvproto/pkg/metapb"
 	"github.com/pingcap/ticdc/cdc/model"
-	"github.com/pingcap/ticdc/pkg/util/testleak"
+	"github.com/stretchr/testify/require"
+
```
Comment (overvenus) [inline pkg/version/check_test.go:29]:
```suggestion
	"github.com/stretchr/testify/require"
```

**Review by overvenus — APPROVED** [review]

File: pkg/version/check_test.go [hunk 2]
```diff
@@ -83,9 +76,9 @@ func (s *checkSuite) TestCheckClusterVersion(c *check.C) {
 		if err == nil {
 			break
 		}
-		c.Error(err)
+		fmt.Fprintf(os.Stderr, "[ERROR]:%v", err)
```
Comment (hi-rustin) [inline pkg/version/check_test.go:79]: How about we use `t.Log(f)`?

Also the original c.Error should make the case fail straight away.

**Review by hi-rustin — COMMENTED** [review]
reset LGTM

File: pkg/version/check_test.go (hunk 2 above)
Comment (maxshuang) [reply pkg/version/check_test.go:79]: (1) t.Log(f) LGTM.
(2) c.Error will fail the test but not stop the test immediately.  So i can't find the corresponding function in  testify like c.Error.  testify has fail/failnow/failf function, but they will stop the test-case directly.    Any suggestions? @hi-rustin

File: pkg/version/check_test.go (hunk 1 above)
Comment (maxshuang) [reply pkg/version/check_test.go:29]: ??? what's the difference?

Comment (hi-rustin) [reply pkg/version/check_test.go:29]: @overvenus He should have meant the empty line below.

File: pkg/version/check_test.go (hunk 2 above)
Comment (hi-rustin) [reply pkg/version/check_test.go:79]: If it's all going to fail, then I see nothing wrong with ending it early. @overvenus What do you think?

File: pkg/version/check_test.go [hunk 3]
```diff
@@ -83,9 +76,9 @@ func (s *checkSuite) TestCheckClusterVersion(c *check.C) {
 		if err == nil {
 			break
 		}
```
Comment (maxshuang) [inline pkg/version/check_test.go:78]: so what's the point for sleep in loop to wait?   cause the retry already means fail when call c.Error()

File: pkg/version/check_test.go (hunk 1 above)
Comment (maxshuang) [reply pkg/version/check_test.go:29]: oo

File: pkg/version/check_test.go (hunk 2 above)
Comment (maxshuang) [reply pkg/version/check_test.go:79]: I find the testify.assert.Fail can act just like c.Error().  But the semantic of  assert in testify still confuses me, cause assert means severe mismatch in C/C++.

Comment (hi-rustin) [reply pkg/version/check_test.go:79]: 😆 Cool!

File: pkg/version/check_test.go (hunk 1 above)
Comment (maxshuang) [reply pkg/version/check_test.go:29]: so, it's a problem adding an empty line here?

File: pkg/version/check_test.go [hunk 4]
```diff
@@ -22,22 +22,15 @@ import (
...
 
 	"github.com/coreos/go-semver/semver"
-	"github.com/pingcap/check"
 	"github.com/pingcap/kvproto/pkg/metapb"
 	"github.com/pingcap/ticdc/cdc/model"
-	"github.com/pingcap/ticdc/pkg/util/testleak"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
 	pd "github.com/tikv/pd/client"
```
Comment (hi-rustin) [inline pkg/version/check_test.go:30]:
```suggestion
	"github.com/stretchr/testify/require"
	pd "github.com/tikv/pd/client"
```

File: pkg/version/check_test.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/version/check_test.go:29]: See: https://github.com/pingcap/ticdc/pull/2920#discussion_r716200957

**Review by ben1009 — APPROVED** [review]

File: pkg/version/check_test.go (hunk 1 above)
Comment (amyangfei) [reply pkg/version/check_test.go]: removed

**Pull request merged**